# This notebook contains:

1. Descriptive analysis

# Data types
# Missing values
# Duplicates
# Basic statistics



In [18]:
# import libraries
import pandas as pd
import numpy as np
import os

In [19]:
# assign path and inporting the csv file to a df
listings = pd.read_csv(r'/Users/renataherrera/Documents/CF RH 2023-2024/CF DATA IMMERSION/CF RH A6 Advanced Analytics Vancouver/Data/Original/vanlistings.csv', index_col = False)

In [20]:
#visualizing first 5 rows of listings dataframe
listings.head()

id                                               name  host_id  \
0  13188  Rental unit in Vancouver · ★4.84 · Studio · 2 ...    51466   
1  13221  Rental unit in Vancouver · ★4.73 · 1 bedroom ·...    51634   
2  13358  Condo in Vancouver · ★4.68 · 1 bedroom · 1 bed...    52116   
3  13490  Rental unit in Vancouver · ★4.93 · 1 bedroom ·...    52467   
4  14267  Home in Vancouver · ★4.76 · 1 bedroom · 2 beds...    56030   

        host_name  neighbourhood_group             neighbourhood   latitude  \
0    Ian And Emma                  NaN                Riley Park  49.247730   
1          Amanda                  NaN                Riley Park  49.254890   
2            Lynn                  NaN                  Downtown  49.281174   
3            Iris                  NaN  Kensington-Cedar Cottage  49.256220   
4  Peter & Alison                  NaN  Kensington-Cedar Cottage  49.249220   

    longitude        room_type  price  minimum_nights  number_of_reviews  \
0 -123.105090  Entire home/apt  150.0               2                283   
1 -123.097080  Entire home/apt  120.0              31                 15   
2 -123.125931  Entire home/apt  165.0               1                493   
3 -123.066070  Entire home/apt  150.0              30                101   
4 -123.081390  Entire home/apt  150.0               3                 33   

  last_review  reviews_per_month  calculated_host_listings_count  \
0  2023-11-28               1.68                               1   
1  2017-12-19               0.15                               1   
2  2023-12-11               3.00                               1   
3  2023-11-16               0.66                               1   
4  2021-07-14               0.21                               1   

   availability_365  number_of_reviews_ltm    license  
0                 0                     30  23-156488  
1               363                      0        NaN  
2                 0                     55  22-311727  
3                92                      5        NaN  
4                17                      0  21-156500

In [21]:
# visualizing last 5 rows
listings.tail()

id                                               name  \
6686  1043694259400656638  Rental unit in Vancouver · ★New · 2 bedrooms ·...   
6687  1043785105970341901  Home in Vancouver · ★New · 1 bedroom · 1 bed ·...   
6688  1043860568247128725  Home in Vancouver · ★New · 3 bedrooms · 4 beds...   
6689  1044526386585207604   Rental unit in Delta · ★New · 1 bedroom · 1 bath   
6690  1044548067589994085   Rental unit in Delta · ★New · 1 bedroom · 1 bath   

        host_id host_name  neighbourhood_group             neighbourhood  \
6686  507416955    Edward                  NaN                  Downtown   
6687   51172453     Felix                  NaN  Kensington-Cedar Cottage   
6688  515469496    Trevor                  NaN                  Oakridge   
6689  550599022       Tei                  NaN                  Downtown   
6690  550599022       Tei                  NaN                  Downtown   

       latitude   longitude        room_type  price  minimum_nights  \
6686  49.275730 -123.115480  Entire home/apt  138.0               1   
6687  49.245914 -123.085170  Entire home/apt    NaN               1   
6688  49.227418 -123.136011  Entire home/apt  232.0               2   
6689  49.278704 -123.108883     Private room   75.0              30   
6690  49.278120 -123.108950     Private room   70.0              30   

      number_of_reviews last_review  reviews_per_month  \
6686                  0         NaN                NaN   
6687                  0         NaN                NaN   
6688                  0         NaN                NaN   
6689                  0         NaN                NaN   
6690                  0         NaN                NaN   

      calculated_host_listings_count  availability_365  number_of_reviews_ltm  \
6686                               1                 6                      0   
6687                               1               269                      0   
6688                               2               167                      0   
6689                               2               248                      0   
6690                               2               248                      0   

        license  
6686     Exempt  
6687        NaN  
6688  23-259590  
6689        NaN  
6690        NaN

In [22]:
#check dimensions
listings.shape

(6691, 18)

In [23]:
#check data types
listings.dtypes

id                                  int64
name                               object
host_id                             int64
host_name                          object
neighbourhood_group               float64
neighbourhood                      object
latitude                          float64
longitude                         float64
room_type                          object
price                             float64
minimum_nights                      int64
number_of_reviews                   int64
last_review                        object
reviews_per_month                 float64
calculated_host_listings_count      int64
availability_365                    int64
number_of_reviews_ltm               int64
license                            object
dtype: object

In [24]:
#viewing column names and looking at ones to keep
listings.columns

Index(['id', 'name', 'host_id', 'host_name', 'neighbourhood_group',
       'neighbourhood', 'latitude', 'longitude', 'room_type', 'price',
       'minimum_nights', 'number_of_reviews', 'last_review',
       'reviews_per_month', 'calculated_host_listings_count',
       'availability_365', 'number_of_reviews_ltm', 'license'],
      dtype='object')

In [25]:
#dropping some columns to keep variables needed and neighborhood_group is Nan values
listings.drop(['name', 'host_id', 'host_name', 'neighbourhood_group', 'last_review'], axis = 1, inplace=True)

In [26]:
listings.shape

(6691, 13)

In [27]:
#visualizing data types
listings.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6691 entries, 0 to 6690
Data columns (total 13 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              6691 non-null   int64  
 1   neighbourhood                   6691 non-null   object 
 2   latitude                        6691 non-null   float64
 3   longitude                       6691 non-null   float64
 4   room_type                       6691 non-null   object 
 5   price                           5851 non-null   float64
 6   minimum_nights                  6691 non-null   int64  
 7   number_of_reviews               6691 non-null   int64  
 8   reviews_per_month               5552 non-null   float64
 9   calculated_host_listings_count  6691 non-null   int64  
 10  availability_365                6691 non-null   int64  
 11  number_of_reviews_ltm           6691 non-null   int64  
 12  license                         48

In [28]:
#checking for mixed types
for col in listings.columns.tolist():
    weird = (listings[[col]].applymap(type) != listings[[col]].iloc[0].apply(type)).any(axis = 1)
    if len (listings[weird]) > 0:
        print (col)

license


In [29]:
# changing data types
#changing names to strings that mixed
listings[['license']] = listings[['license']].astype('str')

In [30]:
#checking for mixed types
for col in listings.columns.tolist():
    weird = (listings[[col]].applymap(type) != listings[[col]].iloc[0].apply(type)).any(axis = 1)
    if len (listings[weird]) > 0:
        print (col)

In [31]:
#Finding duplicates
listings_dups = listings[listings.duplicated()]
listings_dups

Empty DataFrame
Columns: [id, neighbourhood, latitude, longitude, room_type, price, minimum_nights, number_of_reviews, reviews_per_month, calculated_host_listings_count, availability_365, number_of_reviews_ltm, license]
Index: []

There are no duplicates for 'listings' df.

In [32]:
# Running basic stats
listings.describe().to_clipboard()
listings.describe()

id     latitude    longitude         price  minimum_nights  \
count  6.691000e+03  6691.000000  6691.000000   5851.000000     6691.000000   
mean   4.383035e+17    49.262609  -123.111897    201.012647       16.129726   
std    4.319116e+17     0.021067     0.037712    546.121150       28.709975   
min    1.318800e+04    49.202960  -123.217730     14.000000        1.000000   
25%    3.334591e+07    49.249820  -123.130779    100.000000        2.000000   
50%    5.943669e+17    49.269090  -123.115390    139.000000        3.000000   
75%    8.777264e+17    49.278991  -123.090032    204.000000       30.000000   
max    1.044548e+18    49.294360  -123.023730  28650.000000      399.000000   

       number_of_reviews  reviews_per_month  calculated_host_listings_count  \
count        6691.000000        5552.000000                     6691.000000   
mean           40.307278           1.814787                        6.933343   
std            72.113204           1.906942                       19.886758   
min             0.000000           0.010000                        1.000000   
25%             2.000000           0.330000                        1.000000   
50%            11.000000           1.030000                        1.000000   
75%            49.000000           2.872500                        3.000000   
max           948.000000          15.820000                      134.000000   

       availability_365  number_of_reviews_ltm  
count       6691.000000            6691.000000  
mean         137.804065              13.731879  
std          126.172125              20.678307  
min            0.000000               0.000000  
25%            5.000000               0.000000  
50%          102.000000               3.000000  
75%          253.000000              20.000000  
max          365.000000             141.000000

Observations of basicc stats include 75% of price values are below 204.00 with a maximum value of 28650.00. Considering Vancouver is expensive, the max value could be investigated further. 

Meanwhile, miniumum nights @75% are 30 nights whereas the max is at 399 nights, an outlier value.

In [33]:
listings.isnull().sum()

id                                   0
neighbourhood                        0
latitude                             0
longitude                            0
room_type                            0
price                              840
minimum_nights                       0
number_of_reviews                    0
reviews_per_month                 1139
calculated_host_listings_count       0
availability_365                     0
number_of_reviews_ltm                0
license                              0
dtype: int64

In [34]:
#Defining path
path = r'/Users/renataherrera/Documents/CF RH 2023-2024/CF DATA IMMERSION/CF RH A6 Advanced Analytics Vancouver'

In [35]:
#checking path
path

'/Users/renataherrera/Documents/CF RH 2023-2024/CF DATA IMMERSION/CF RH A6 Advanced Analytics Vancouver'

In [36]:
listings.to_csv(os.path.join(path, 'Data', 'Prepared Data', 'cleaned_van_listings.csv')) 